plugins/org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/internal/RootResolvedTypes.java

```diff
@@ -167,6 +167,12 @@ protected boolean isIntentionalEarlyExit(/* @Nullable */ XExpression expression)
 	}
 	
 	protected AbstractDiagnostic createTypeDiagnostic(XExpression expression, LightweightTypeReference actualType, LightweightTypeReference expectedType) {
+		if (expectedType.isUnknown() && actualType.isAny())
```

Comment:
This looks wrong rather ad-hoc to me. Any should already be conformant to unknown, otherwise the fix should go into the conformance check.